### PR TITLE
[feat] Add bearer_token to grpc header

### PR DESCRIPTION
**Pull request by jennifer12121**
Related issue: https://github.com/aimhubio/aim/issues/2207

### CODE REVIEW

**Review by alberttorosyan — COMMENTED** [review]
@jennifer12121 thanks a lot for contribution! Nice job 🙌

I'd suggest to change the implementation a bit to avoid the need of passing the `bearer_token` through multiple objects `__init__`s.
A couple of possible options:
- add a `classmethod` to the `class Client` to set token directly (without passing it to Repo/Run classes)
- read token value from pre-defined environment variable or file
- introduce a new Client config. accessible from any part of the code and set the token value as a config option.

**Comment by jennifer12121** [comment]
Accidentally closed while updating to the latest version. Reopening.

**Comment by jennifer12121** [comment]
Hi @alberttorosyan , apologies on the delay on this! I took your second suggestion to create a new environment variable called `AIM_RT_BEARER_TOKEN` and I only pass that into the `create_remote_channels` method inside of `Client`. Thank you for your feedback and please let me know if you have any questions/comments!

**Comment by alberttorosyan** [comment]
Hey @jennifer12121!
Thanks for applying the changes and sorry for the delay. Will review the changes once again and get back to you 🙌

File: aim/ext/transport/client.py [hunk 1]
```diff
@@ -76,15 +77,20 @@ def _create_remote_channels(self):
 
         ssl_certfile = os.getenv(AIM_CLIENT_SSL_CERTIFICATES_FILE)
         msg_max_size = int(os.getenv(AIM_RT_MAX_MESSAGE_SIZE, AIM_RT_DEFAULT_MAX_MESSAGE_SIZE))
+        bearer_token = os.getenv(AIM_RT_BEARER_TOKEN)
```
Comment (mihran113) [inline aim/ext/transport/client.py:80]: Won't this just fail due to the fact that: 
```
AIM_RT_BEARER_TOKEN = None
```

File: aim/ext/transport/config.py [hunk 2]
```diff
@@ -11,3 +11,4 @@
 # GRPC OPTIONS
 AIM_RT_MAX_MESSAGE_SIZE = '__AIM_RT_MAX_MESSAGE_SIZE__'
 AIM_RT_DEFAULT_MAX_MESSAGE_SIZE = 16 * 1024 * 1024  # 16MB
+AIM_RT_BEARER_TOKEN = None
```
Comment (mihran113) [inline aim/ext/transport/config.py:14]: I guess this should look like something like this: 
```
AIM_RT_BEARER_TOKEN = '__AIM_RT_BEARER_TOKEN__'
```

File: aim/ext/transport/client.py [hunk 3]
```diff
@@ -76,15 +77,20 @@ def _create_remote_channels(self):
...
         options = [
             ('grpc.max_send_message_length', msg_max_size),
             ('grpc.max_receive_message_length', msg_max_size)
         ]
 
         # open a channel with router
         if ssl_certfile:
```
Comment (mihran113) [inline aim/ext/transport/client.py:87]: this check also needs to be changed, so if the `bearer_token` is provided then we should go with the case of secure channel as well

File: aim/ext/transport/client.py [hunk 4]
```diff
@@ -76,15 +77,20 @@ def _create_remote_channels(self):
...
+            self._call_credentials = grpc.access_token_call_credentials(bearer_token)
```
Comment (mihran113) [inline aim/ext/transport/client.py:88]: Let's add an explicit check that bearer token exists, and if so, then add the credentials:
```
if bearer_token:
    self._call_credentials = grpc.access_token_call_credentials(bearer_token)
    root_certificates = grpc.composite_channel_credentials(
         root_certificates, self._call_credentials
     )
```

**Review by mihran113 — CHANGES_REQUESTED** [review]

File: aim/ext/transport/config.py (hunk 2 above)
Comment (jennifer12121) [reply aim/ext/transport/config.py]: Thank you for this feedback! I made the change you suggested.

**Pull request merged**